### DISCUSSION TITLE
Prevent double detection / object splitting

### DISCUSSION THREAD

**Discussion by AlainPilon** [roboflow/supervision#450] [Q&A]
Our current setup is to run components on a conveyor belt with an overhead camera doing the detection/counting. The components are far enough apart from each other to never create any interferences. 

In some cases, the component is detected as being 2 parts for a short amount of time, creating issues with the counting. 

What is the typical way to handle these cases? Is there a functionality in Supervision to handle this?

**Comment by SkalskiP** [comment]
Is there a significant overlap between those two detections? Can you share an example of the frame where that occurs?

**Reply by AlainPilon** [reply]
Here are two examples. 

In this case, there is a complete overlap (I really dont understand what is going on there to create 2 detections!):

![12_frame](https://github.com/roboflow/supervision/assets/37088/8b0ca898-5e0e-4b81-8e6e-62ef028ab1c8)

In this case there is an intersection (our model has not seen enough of these big parts to understand that they are a single component):
![1_frame](https://github.com/roboflow/supervision/assets/37088/8ba21511-37ff-4ca8-ac07-3b77b242325a)

There is also a third case where the two boxes are close to each others but without overlapping. 

I feel like it should be SV responsibility to merge these together. Obviously, this behaviour should be a setting as not everyone can ensure all subjects are never overlapping. 


BTW, going to upgrade to Supervision 0.15 this afternoon to get rid of these ugly red annotation boxes!:-)

**Reply by SkalskiP** [reply]
One thing that we have now is [with_nms](https://supervision.roboflow.com/detection/core/#supervision.detection.core.Detections.with_nms) method. That will remove double detections. But the problem is it is not merging but rather removing extra detections. So In the case of the first image, that is probably good, but in the case of the second, probably not.

**Reply by SkalskiP** [reply]
One more thing. Some time ago, @hardikdava proposed adding this https://github.com/roboflow/supervision/issues/268. I believe it would be the most useful for you.

**Reply by AlainPilon** [reply]
The WBF looks interesting (based on code sample in the article) as it is a more robust version of what I had in mind for our specific use case. 

My idea was to look at all the detection with the same class and merge them together if they were closer than X apart. Given components are equality spaced out on the conveyor belt, it should be sufficient. 

Thanks a lot for the pointer!
